packages/messaging/src/Message.utils.ts

```diff
@@ -15,12 +15,41 @@ import {
   CompressedAttestedClaim,
   CompressedMessageBody,
   MessageBody,
+  CompressedRequestClaimsForCTypesContent,
+  IRequestClaimsForCTypesContent,
+  ICType,
 } from '@kiltprotocol/types'
 import { SDKErrors } from '@kiltprotocol/utils'
 import Message from '.'
 
 /**
- * [STATIC] Compresses a [[Message]] depending on the message body type.
+ * Verifies required properties for a given [[CType]] before sending or receiving a message.
+ *
+ * @param requiredProperties The list of required properties that need to be verified against a [[CType]].
+ * @param cType A [[CType]] used to verify the properties.
+ * @throws [[ERROR_CTYPE_HASH_NOT_PROVIDED]] when the properties do not match the provide [[CType]].
+ *
+ * @returns Returns the properties back.
+ */
+
+export function verifyRequiredCTypeProperties(
+  requiredProperties: string[],
+  cType: ICType
+): boolean {
+  if (!cType.hash) throw SDKErrors.ERROR_CTYPE_HASH_NOT_PROVIDED()
```

Comment:
Shouldn't we use our isICType typeguard instead of just checking the hash?

Reply:
I agree, I have updated it.